zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -321,6 +326,21 @@ public void saveToFile() throws IOException {
     configStorage.save(info);
   }
 
+  private void initMetrics() {
+    // This class should  be a singleton
+    // Tests create more instances
+    Metrics.getMetricRegistry().removeMatching(
+      MetricFilter.startsWith(MetricRegistry.name(METRICS_PREFIX_INTERPRETER)));
+    Metrics.getMetricRegistry().<Gauge<Integer>>register(MetricRegistry.name(METRICS_PREFIX_INTERPRETER, "amount"),
+      interpreterSettings::size);
+    Metrics.getMetricRegistry().<Gauge<Integer>>register(MetricRegistry.name(METRICS_PREFIX_INTERPRETER, "group", "size"),
+      () -> getAllInterpreterGroup().size());
+    for (Map.Entry<String, InterpreterSetting> interpreterSetting : interpreterSettings.entrySet()) {
```

Comment:
Not dynamic, we should register the metric when the interpreter is created and remove the registration when the interpreter is removed.